### PR TITLE
[threadpool-ms] Exit worker thread after 5 to 60 seconds

### PR DESCRIPTION
**Pull request by ludovic-henry**
The timeout before exiting a worker thread is random to limit the worst case scenario where, if the timeout is set to N seconds, the user would launch a lot of work item simultaneously, at an interval of N + 1 seconds. In this case, we would have to kill all the threads at once, and starts them all over again 1 second later. The randomization makes the creation and destruction more linear in time, and avoid burst.

### CODE REVIEW

**Comment by Therzok** [comment]
@alanmcgovern will be happy!

File: mono/metadata/threadpool-ms.c [hunk 1]
```diff
@@ -501,10 +537,12 @@ worker_park (void)
 	mono_mutex_lock (&threadpool->active_threads_lock);
 
 	if (!mono_runtime_is_shutting_down ()) {
+		MonoInternalThread *thread = mono_thread_internal_current ();
```
Comment (kumpera) [inline mono/metadata/threadpool-ms.c]: This variable is not used anywhere

Comment (ludovic-henry) [inline mono/metadata/threadpool-ms.c]: It is, on line 543 and 547.

**Comment by kumpera** [comment]
build

**Pull request closed without merging**